src/Symfony/Component/Security/Core/Authentication/Token/SwitchUserToken.php

```diff
@@ -19,39 +19,47 @@
 class SwitchUserToken extends UsernamePasswordToken
 {
     private $originalToken;
+    private $originatedFromUri;
 
     /**
-     * @param string|object $user        The username (like a nickname, email address, etc.), or a UserInterface instance or an object implementing a __toString method
-     * @param mixed         $credentials This usually is the password of the user
+     * @param string|object $user              The username (like a nickname, email address, etc.), or a UserInterface instance or an object implementing a __toString method
+     * @param mixed         $credentials       This usually is the password of the user
+     * @param string|null   $originatedFromUri The URI where was the user at the switch
      *
      * @throws \InvalidArgumentException
      */
-    public function __construct($user, $credentials, string $firewallName, array $roles, TokenInterface $originalToken)
+    public function __construct($user, $credentials, string $firewallName, array $roles, TokenInterface $originalToken, string $originatedFromUri = null)
     {
         parent::__construct($user, $credentials, $firewallName, $roles);
 
         $this->originalToken = $originalToken;
+        $this->originatedFromUri = $originatedFromUri;
     }
 
     public function getOriginalToken(): TokenInterface
     {
         return $this->originalToken;
     }
 
+    public function getOriginatedFromUri(): ?string
+    {
+        return $this->originatedFromUri;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function __serialize(): array
     {
-        return [$this->originalToken, parent::__serialize()];
+        return [$this->originalToken, $this->originatedFromUri, parent::__serialize()];
```

Comment:
Don't we need to add at the end to avoid issues with existing data when applications are updated?